proposals/permissions-requestSiteAccess-api.md

```diff
@@ -0,0 +1,135 @@
+
+#### Other Alternatives considered
+
+Action API is used to control the extension’s button in the browser’s toolbar. It’s exposed if the extension includes the "action" key in the manifest. This is troublesome since an extension could not have an action, but still want to show site access requests. We should not limit requests for extensions with actions.
+
+### Manifest Changes
+
+None.
+
+### Custom Bindings
+
+None.
+
+### Design and Implementation
+
+#### Persistence
+
+Showing a site access request is never persisted.
+
+#### Alignment with Other Vendors and Action APIs
+
+Each browser can decide how they want to signal the extension site access requests. Browsers may decide to ignore the request (e.g due to noisiness, or if an extension has no visible UI).
+On Chrome’s side, we are exploring adding an ‘Allow’ chip in the extensions toolbar.
+
+## User Experience
+
+### UI Elements and User Visible-Effects
+
+Each browser will handle the request as desired. On Chrome’s side, we are considering showing an ‘Allow’ button in the toolbar for these requests. On click, it would grant persistent access to the site’s origin (which can be again withheld by the user)
+
+!["Site access request on Chrome toolbar"](./assets/permissions-requestSiteAccess-chrome-toolbar.png)
+
+We would also provide a way for users to silence the requests for an extension through a setting in the extensions menu and in chrome://extensions.
+
+!["Extensions menu on Chrome"](./assets/permissions-requestSiteAccess-chrome-menu.png)
+
+#### Attribution
+The site access request will be attributed directly to the extension.
+
+## Security and Privacy
+
+### Exposed Sensitive Data
+
+This API does not directly expose any sensitive data to the extension. However, it could lead to the extension gaining access to the site.
+
+### Abuse Scenarios
+
+Extensions can enable site access requests on every tab, which could be annoying to the user.
+
+#### Mitigations
+Attacks of annoyance are not in our threat model. Extensions can have annoying behaviors, but they risk uninstallation.
+On the browser side, we can place some restrictions to limit noisiness:
+- User can decide whether they want to see site access requests for an specific extension
+- We can explore setting a max amount of times the request is shown when the user ignores it.
+
+### Additional Security Considerations
+None
+
+## Alternatives
+
+### Existing Workarounds
+
+Currently, extensions have site access by default. Users can change the site access to be when clicked. The only way for extensions to signal the user they want access to the site is through permissions.request() flow which is noisy and requires a user gesture. Otherwise, they rely on the user realizing through the browser UI (e.g extensions menu).
```

Comment:
> Currently, extensions have site access by default. Users can change the site access to be when clicked.

This is browser specific, and not true for all browsers. In Safari extension don't have any site access by default.

Reply:
Thanks for the callout, I reworded it to be more browser inclusive